noxfile.py

```diff
@@ -230,3 +249,47 @@ def _get_rust_target() -> str:
 
 
 _HOST_LINE_START = "host: "
+
+
+def _get_coverage_env() -> Dict[str, str]:
+    env = {}
+    output = subprocess.check_output(["cargo", "llvm-cov", "show-env"], text=True)
+
+    for line in output.strip().splitlines():
+        (key, value) = line.split("=", maxsplit=1)
+        env[key] = value.strip('"')
+
+    # Ensure that examples/ and pytests/ all build to the correct target directory to collect
+    # coverage artifacts.
+    env["CARGO_TARGET_DIR"] = env["CARGO_LLVM_COV_TARGET_DIR"]
+
+    return env
+
+
+def _run(session: nox.Session, *args: str, **kwargs: Any) -> None:
+    """Wrapper for _run(session, which creates nice groups on GitHub Actions."""
+    if "GITHUB_ACTIONS" in os.environ:
+        # Insert ::group:: at the start of nox's command line output
+        print("::group::", end="", flush=True)
+    session.run(*args, **kwargs)
+    if "GITHUB_ACTIONS" in os.environ:
+        print("::endgroup::")
```

Comment:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/1556054/191687273-0d63a1a3-9a2a-4abf-901f-8c9d3fbfd787.png">

https://github.com/PyO3/pyo3/actions/runs/3103635220/jobs/5027187965#step:7:10

Looks odd, perhaps `::group::` should have a description like `::group::Running xxx`?

Reply:
It should be using nox's output from the next line but I think it's because nox's command output goes to stderr it's not lining up correctly in CI. I'm going to try printing to stderr.

Reply:
Yep that looks like it works much better now.